### DISCUSSION TITLE
Using  Signac on HPC

### DISCUSSION THREAD

**Discussion by Melody5528** [stuart-lab/signac#1075] [Q&A] [answered]
Does anyone have experience installing Signac on the HPC server? I am having so much trouble downloading the updated full package and its dependencies.  The same exact codes worked perfectly fine on a local PC which unfortunately is too slow. 

Any suggestions or recommendations would be greatly appreciated!!

**Comment by timoast** [comment, marked as answer]
Using the conda package might be easier, since it installs precompiled binaries: https://satijalab.org/signac/articles/install.html#conda

**Reply by Melody5528** [reply]
Thank you very much. The Conda worked.

Our IT helped me figure out the problem with the "Signac" package installation in RStudio. It requires an updated "Matrix" program, but the default path led to an older version, preventing the "Signac" download. 

The temporary remedy is to install these two packages manually, 

BiocManager::install("GenomeInfoDbData")
BiocManager::install("GenomeInfoDb")

Then proceed with the "Signac" installation again. This time, a current version of "matrix" was picked up, and the installation was finished without further issues.

I hope this short article will help others with the same problems.

Please update the default path of "Matrix" in the "Signac" package. 

Thank you again!!!

**Reply by timoast** [reply]
Installation of `GenomeInfoDbData` often causes issues because it's hosted in the BioC annotation repository rather than BioC software. Running `setRepositories(ind=1:3)` should enable it to be automatically installed as needed.
